src/mongo/util/IndexUtils.class.php

```diff
@@ -65,10 +65,14 @@ public function ensureIndexes($reindex=false,$storeName=null,$background=true)
             // Index views
             foreach($config->getViewSpecifications($storeName) as $viewId=>$spec)
             {
-                $collection = Config::getInstance()->getCollectionForView($storeName, $viewId);
+                $collection = $config->getCollectionForView($storeName, $viewId);
                 if($collection)
                 {
-                    $indexes = array(array("_id.type"=>1));
+                    $indexes = array(
+                        array(_ID_KEY.'.'._ID_RESOURCE=>1, _ID_KEY.'.'._ID_CONTEXT=>1, _ID_KEY.'.'._ID_TYPE=>1),
+                        array(_ID_KEY.'.'._ID_TYPE=>1),
+                        array("value."._IMPACT_INDEX=>1)
```

Comment:
is there no constant for `"value.impactIndex"`?  Surprised we haven't got anything already!